### DISCUSSION TITLE
Comments and questions about v2

### DISCUSSION THREAD

**Discussion by cbelena** [jpanther/congo#116] [General]
First of all, thank you very much for your work continuing to improve this fantastic theme, @jpanther!

I have tested it on my yet-to-be-published new personal page and I have some comments/suggestions/questions about version 2:

1. I noticed there are new subtle animations when the cursor moves above links, that's nice! However, in some links I have found that the colour of the underline changes from black/white (depends on appearance) to the correct theme's colour. Is that a design choice or a bug? For example, in `exampleSite`:
![image](https://user-images.githubusercontent.com/97797388/153770952-f7a40ff6-ffeb-4a72-a332-fe9d7da100e2.png)
...and 150ms later:
![image](https://user-images.githubusercontent.com/97797388/153770981-a084263f-50c4-4e7b-b0f2-52009bc74644.png)
2. The new `headline` is great! But I find it a little too close to the title. Maybe add some padding?
How appears now:
![image](https://user-images.githubusercontent.com/97797388/153771551-64ad7d11-d8f3-435a-985e-80510dfa5a26.png)
...and with 0.25rem padding (top and bottom):
![image](https://user-images.githubusercontent.com/97797388/153771583-b27bb777-5a84-4393-ad73-1500caa23b97.png)
3. When I was translating the new words added in version 2, I always like to understand their context and find where are they being used. And then found "_Skip to main content_". What's its use? I realize it's in `baseof.html`, and allows to jump to `#main-content`, but... how to use it? Am I missing something? 🤔
4. I fully understand why you changed the resources of the theme (author image and site logo) to be `assets` instead of `static` images. But, in my case, I was already using optimized author image, and now the theme reprocesses it, with everything it implies. I'm probably in the minority, but I think it would be great to include a setting to allow processing these two resources (default) or leave them as is (v1.x behaviour). And yes, I also don't like the _very ugly_ URIs Hugo generates to the processed assets! 😅😅😅
5. I initially thought that the main content area not growing to fill screen vertically was a new design choice for v2, but it was reverted last realease so nothing to say here 😁
6. Speaking of filling the screen, I have a very brief home page, so on mobile there should be no scrolling. But I noticed that on iOS the navbar of Safari was blocking the footer, and you had to scroll to reveal it (I think this was also happening on version 1.x.). I have [read](https://allthingssmitty.com/2020/05/11/css-fix-for-100vh-in-mobile-webkit/) about this problem; it seems it was designed that way. But I didn't like it, so I added this to `custom.css`:
    ```css
    .h-screen {
      max-height: -webkit-fill-available;
    }
    ```
    ...and now it shows the full page without scrolling. I don't know if this is the cleanest way to do it, but it works for me (well, not in Android... 😕) and wanted to share it. 

Well, that's it... for now! 😂

**Comment by jpanther** [comment]
Thanks for the useful feedback. Always good to hear what is and isn't working well. To address each of your points...

1. This is not correct behaviour. What browser are you using? I do most of my testing in Webkit/Safari and don't see this happening. I have a feeling it's Firefox that's doing this.
2. I'll take another look at this. More padding (almost) never hurts.
3. Skip to main content is assistive for keyboard-only users. You have to use the select next link keyboard shortcut for your browser to make it appear (in Safari, for instance, it's Option + Tab, then spacebar to activate the link). You can read more about why these links are helpful for some users here: https://webaim.org/techniques/skipnav/
4. Probably not high on my priorities right now. The reason I started doing this was to achieve Lighthouse benchmarks. One of the issues with the author image is that it needs to be different sizes on the homepage and on the author footer. Providing one asset does not let you optimise for both uses.
5. Yes, this was a bug.
6. iOS Safari is notorious for these types of layout issues. I'll look into this a bit more and see if it's a workable solution. Seems strange that Tailwind doesn't do this already so maybe there's a catch.

**Reply by cbelena** [reply]
>This is not correct behaviour. What browser are you using? I do most of my testing in Webkit/Safari and don't see this happening.

You're right, Safari on macOS goes from no underline to the tinted underline. But on different browsers, such Chrome (tested on macOS and Windows) or Firefox, I'm seeing what I described on my OP. It's very subtle because it changes very fast (150ms).

>Skip to main content is assistive for keyboard-only users. [...]

Thanks for clarifying, now it makes sense! 👍

>One of the issues with the author image is that it needs to be different sizes on the homepage and on the author footer. Providing one asset does not let you optimise for both uses.

Yeah, that's a valid point. But it also has one _little drawback_: if an user navigates from homepage to a post (or vice versa) it has to load two different images instead of only one. I know, I know... 😅

Thank you again!! 😄

**Reply by jpanther** [reply]
I've opened https://github.com/jpanther/congo/issues/125 to track the issue with the underline styles.

**Reply by cbelena** [reply]
Thank you!
